pkg/reconciler/function/function.go

```diff
@@ -102,8 +103,17 @@ func (r *Reconciler) ReconcileKind(ctx context.Context, o *functionv1alpha1.Func
 	}
 	o.Status.MarkConfigmapAvailable()
 
+	sink, err := r.resolveSink(ctx, o)
+	if err != nil {
+		logger.Error("Error resolving sink", zap.Error(err))
+		o.Status.MarkSinkUnavailable()
+		return err
+	}
```

Comment:
This will cause a double logging of the same error, on every error:
* once here
* once after the call to the top-level `ReconcileKind()`, which bubbles up the error at line 110

Suggestion to avoid that: simply return the error wrapped with a short context string:

```suggestion
	sink, err := r.resolveSink(ctx, o)
	if err != nil {
		o.Status.MarkSinkUnavailable()
		return fmt.Errorf("resolving sink URI: %w", err)
	}
```